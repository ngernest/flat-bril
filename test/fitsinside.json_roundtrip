{
  "functions": [
    {
      "args": [
        {
          "name": "width1",
          "type": "int"
        },
        {
          "name": "height1",
          "type": "int"
        },
        {
          "name": "width2",
          "type": "int"
        },
        {
          "name": "height2",
          "type": "int"
        }
      ],
      "instrs": [
        {
          "args": [
            "width1",
            "height1",
            "width2",
            "height2"
          ],
          "dest": "output",
          "funcs": [
            "fitsInside"
          ],
          "labels": [],
          "op": "call",
          "type": "bool"
        },
        {
          "args": [
            "output"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        }
      ],
      "name": "main"
    },
    {
      "args": [
        {
          "name": "w1",
          "type": "int"
        },
        {
          "name": "h1",
          "type": "int"
        },
        {
          "name": "w2",
          "type": "int"
        },
        {
          "name": "h2",
          "type": "int"
        }
      ],
      "instrs": [
        {
          "args": [
            "w1",
            "w2"
          ],
          "dest": "width_check",
          "funcs": [],
          "labels": [],
          "op": "le",
          "type": "bool"
        },
        {
          "args": [
            "h1",
            "h2"
          ],
          "dest": "height_check",
          "funcs": [],
          "labels": [],
          "op": "le",
          "type": "bool"
        },
        {
          "args": [
            "width_check",
            "height_check"
          ],
          "dest": "first_check",
          "funcs": [],
          "labels": [],
          "op": "and",
          "type": "bool"
        },
        {
          "args": [
            "w1",
            "h2"
          ],
          "dest": "widthheight_check",
          "funcs": [],
          "labels": [],
          "op": "le",
          "type": "bool"
        },
        {
          "args": [
            "h1",
            "w2"
          ],
          "dest": "heightwidth_check",
          "funcs": [],
          "labels": [],
          "op": "le",
          "type": "bool"
        },
        {
          "args": [
            "widthheight_check",
            "heightwidth_check"
          ],
          "dest": "second_check",
          "funcs": [],
          "labels": [],
          "op": "and",
          "type": "bool"
        },
        {
          "args": [
            "first_check",
            "second_check"
          ],
          "dest": "ret_val",
          "funcs": [],
          "labels": [],
          "op": "or",
          "type": "bool"
        },
        {
          "args": [
            "ret_val"
          ],
          "funcs": [],
          "labels": [],
          "op": "ret"
        }
      ],
      "name": "fitsInside",
      "type": "bool"
    }
  ]
}
