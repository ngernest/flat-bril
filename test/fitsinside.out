@main
	op: call 		dest: output		type: bool		args: ["width1", "height1", "width2", "height2"]		funcs: ["fitsInside"]
	op: print		args: ["output"]	
@fitsInside
	op: le   		dest: width_check		type: bool		args: ["w1", "w2"]	
	op: le   		dest: height_check		type: bool		args: ["h1", "h2"]	
	op: and  		dest: first_check		type: bool		args: ["width_check", "height_check"]	
	op: le   		dest: widthheight_check		type: bool		args: ["w1", "h2"]	
	op: le   		dest: heightwidth_check		type: bool		args: ["h1", "w2"]	
	op: and  		dest: second_check		type: bool		args: ["widthheight_check", "heightwidth_check"]	
	op: or   		dest: ret_val		type: bool		args: ["first_check", "second_check"]	
	op: ret  		args: ["ret_val"]	
